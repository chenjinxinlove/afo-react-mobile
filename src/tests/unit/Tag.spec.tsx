import { shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Tag from '../../pageages/tag/';
import DemoTag from '../../pageages/tag/demo';

describe('Button', () => {
  // 测试props type
  it('props type', () => {
    const wrapper = shallow(<Tag
      type = 'primary'
      plain = {true}
    />)
    expect(wrapper.hasClass('afo-tag--primaryplain')).toBe(true)
    expect(wrapper.hasClass('afo-tag')).toBe(true)
  })
  // 测试props mark
  it('props mark', () => {
    const wrapper = shallow(<Tag
      mark = {true}
      className = 'test'
    />)
    expect(wrapper.hasClass('afo-tag--mark')).toBe(true)
    expect(wrapper.hasClass('test')).toBe(true)
  })
  // 测试快照
  it('test snapshots', () => {
    const component = renderer.create(
      <DemoTag />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
 