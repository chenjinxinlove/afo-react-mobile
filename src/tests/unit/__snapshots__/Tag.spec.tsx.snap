// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test snapshots 1`] = `
<div>
  <div
    className="show"
  >
    <div
      className="afo-tag afo-tag--default"
      style={undefined}
    >
      标签
    </div>
    <div
      className="afo-tag afo-tag--primary"
      style={undefined}
    >
      标签
    </div>
    <div
      className="afo-tag afo-tag--danger"
      style={undefined}
    >
      标签
    </div>
  </div>
  <div
    className="show"
  >
    <div
      className="afo-tag afo-tag--defaultplain"
      style={undefined}
    >
      标签
    </div>
    <div
      className="afo-tag afo-tag--primaryplain"
      style={undefined}
    >
      标签
    </div>
    <div
      className="afo-tag afo-tag--dangerplain"
      style={undefined}
    >
      标签
    </div>
  </div>
  <div
    className="show"
  >
    <div
      className="afo-tag afo-tag--default afo-tag--mark"
      style={undefined}
    >
      标签
    </div>
    <div
      className="afo-tag afo-tag--primary afo-tag--mark"
      style={undefined}
    >
      标签
    </div>
    <div
      className="afo-tag afo-tag--danger afo-tag--mark"
      style={undefined}
    >
      标签
    </div>
  </div>
</div>
`;
